src/harness/reference_models/dpa/dpa_mgr.py

```diff
@@ -481,55 +513,100 @@ def _CalcTestPointInterfDiff(point,
   using the US border (with a buffer) as the delimitation.
+  Approximate minimum distance can be specified to avoid too many points for small
+  inland DPAs. Single point DPA are obviously protected by a single point.
 
   Args:
-    num_pts_front_border: number of points in the front border
-    num_pts_back_border: number of points in the back border
-    num_pts_front_zone: number of points in the front zone
-    num_pts_back_zone: number of points in the back zone
-    front_us_border_buffer_km: buffering of US border for delimiting front/back.
+    dpa_geometry: The DPA geometry as a |shapely.Polygon or Point|.
+    num_pts_front_border: Number of points in the front border.
+    num_pts_back_border: Number of points in the back border.
+    num_pts_front_zone: Number of points in the front zone.
+    num_pts_back_zone: Number of points in the back zone.
+    front_us_border_buffer_km: Buffering of US border for delimiting front/back.
+    min_dist_front_border_pts_km: Minimum distance between front border points (km).
+    min_dist_back_border_pts_km: Minimum distance between back border points (km).
+    min_dist_front_zone_pts_km: Minimum distance between front zone points (km).
+    min_dist_back_zone_pts_km: Minimum distance between back zone points (km).
   """
-  us_border = zones.GetUsBorder()
-  dpa_zone = zones.GetDpaZones()[name]
-  # Case of DPA points
-  if isinstance(dpa_zone, sgeo.Point):
-    return [ProtectionPoint(longitude=dpa_zone.x,
-                            latitude=dpa_zone.y)]
-
-  us_border_ext = us_border.buffer(front_us_border_buffer_km / 111.)
-  front_border = dpa_zone.exterior.intersection(us_border_ext)
-  back_border = dpa_zone.exterior.difference(us_border_ext)
-  front_zone = dpa_zone.intersection(us_border_ext)
-  back_zone = dpa_zone.difference(us_border_ext)
-
-  step_front_dpa_arcsec = np.sqrt(front_zone.area / num_pts_front_zone) * 3600.
-  step_back_dpa_arcsec = np.sqrt(back_zone.area / num_pts_back_zone) * 3600.
-  def SampleLine(line, num_points):
+  def SampleLine(line, num_points, min_step_arcsec):
     step = line.length / (num_points-1)
-    return [line.interpolate(dist) for dist in np.arange(0, line.length, step)]
-  return (
-      [ProtectionPoint(longitude=pt.x, latitude=pt.y)
-       for pt in SampleLine(front_border, num_pts_front_border)] +
-      [ProtectionPoint(longitude=pt.x, latitude=pt.y)
-       for pt in SampleLine(back_border, num_pts_back_border)] +
-      [ProtectionPoint(longitude=pt[0], latitude=pt[1])
-       for pt in utils.GridPolygon(front_zone, step_front_dpa_arcsec)] +
-      [ProtectionPoint(longitude=pt[0], latitude=pt[1])
-       for pt in utils.GridPolygon(back_zone, step_back_dpa_arcsec)])
+    min_step = min_step_arcsec / 3600.
+    if step < min_step:
+      num_points = round(line.length / min_step) + 1
+      step = line.length / (num_points-1)
+    return [line.interpolate(dist) for dist in np.arange(0, line.length+step/2., step)]
```

Comment:
Yes you are right. It will create a duplicate if one of the line loop around (as is the case for . It will also create some duplicate if we have the back line not empty (like for coastal DPA), as when we sample the back line we will restart from the last point of the front line.

So I am changing back the condition to not include the last point of a line, since it will be captured in both cases anyway. And changing back the condition:
```
  step=line.length / (num_points-1)
into:
  step = line.length / num_points
```
Hence we will have the correct number of requested points.